cmd/syft/internal/test/integration/go_compiler_detection_test.go

```diff
@@ -12,14 +12,14 @@ func TestGolangCompilerDetection(t *testing.T) {
 		name              string
 		image             string
 		expectedCompilers []string
-		expectedCPE       []cpe.CPE
+		expectedCPE       []cpe.SourcedCPE
```

Comment:
maybe `AnnotatedCPE` since this would not preclude folks from adding more fields that are not automatically out of scope of the single "source" field.

Reply:
Or change `cpe.CPE` to `cpe.Attributes` and `cpe.SourcedCPE` to `cpe.CPE`